web/src/SidebarItem.tsx

```diff
@@ -34,7 +34,7 @@ class SidebarItem {
   queued: boolean
   lastBuild: Build | null = null
   hold: Hold | null = null
-  disabled: boolean
+  disabled: boolean // TODO (lizz): Refactor sidebar logic to refer to other status fields, since this is redundant
```

Comment:
FWIW, I suspect we should find a way to use this class (or something like it) for the table view so that we don't keep having to implement everything twice

Reply:
i totally agree. it would be nice to have a common interface that takes a `UIResource` and returns a view-agnostic instance.